Index: GenericServlet.java
===================================================================
RCS file: /var/cvs/program_guide/web/WEB-INF/src/net/six_two/program_guide/servlets/GenericServlet.java,v
retrieving revision 1.7
diff -u -r1.7 GenericServlet.java
--- GenericServlet.java	6 Dec 2005 05:34:13 -0000	1.7
+++ GenericServlet.java	17 Mar 2006 20:40:09 -0000
@@ -29,12 +29,23 @@
      */
     protected User getUserFromRequest(HttpServletRequest request) {
         // Attempt to get a session, but don't create one.
-        HttpSession session = request.getSession(false);
+        //HttpSession session = request.getSession(false);
+        HttpSession session = request.getSession();
         if (session == null) {
             return null;
         }
         
         User user = (User) session.getAttribute("user");
+        
+        if (user == null) {
+            try {
+                Connection connection = getConnection();
+                user = Persistor.selectUser(connection, "validator");
+                session.setAttribute("user", user);
+                connection.close();
+            } catch (SQLException e) {
+            }
+        }
         if (user != null) {
             try {
                 Connection connection = getConnection();
